tests/test_personal_names.py

```diff
@@ -44,56 +44,65 @@ def test_display_name_to_sort_name(self):
         # Make sure the sort name algorithm processes the messy reality of contributor
         # names in a way we expect.
 
+        m = display_name_to_sort_name
+
         # no input means don't do anything
-        sort_name = display_name_to_sort_name(None)
+        sort_name = m(None)
         eq_(None, sort_name)
 
-        # already sort-ready input means don't do anything
-        sort_name = display_name_to_sort_name(u"Bitshifter, Bob")
-        eq_(u"Bitshifter, Bob", sort_name)
-
-        sort_name = display_name_to_sort_name(u"Prince")
-        eq_(u"Prince", sort_name)
-
-        sort_name = display_name_to_sort_name(u"Pope Francis")
-        eq_(u"Pope, Francis", sort_name)
-
-        sort_name = display_name_to_sort_name(u"Bob Bitshifter")
+        def unchanged(x):
+            # Verify that the input is already a sort name -- either
+            # because it's in "Family, Given" format or for some other
+            # reason.
+            eq_(x, m(x))
+        unchanged(u"Bitshifter, Bob")
+        unchanged(u"Prince")
+        unchanged(u"Pope Francis")
+        unchanged(u"Lord Byron")
+        unchanged(u"Heliodorus (bp. of Tricca.)")
+        unchanged(u"谢新源 (Xie Xinyuan)")
```

Comment:
These four names were treated incorrectly by the old code -- the sort name contained an extra comma when it should have been left alone.